coral/src/app/layout/header/HeaderMenuLink.test.tsx

```diff
@@ -1,61 +1,101 @@
+import data from "@aivenio/aquarium/dist/src/icons/console";
+import { cleanup, render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
 import HeaderMenuLink from "src/app/layout/header/HeaderMenuLink";
-import { cleanup, screen, render, within } from "@testing-library/react";
-import data from "@aivenio/design-system/dist/src/icons/console";
+import { tabNavigateTo } from "src/services/test-utils/tabbing";
 
 // mock out svgs to avoid clutter
-jest.mock("@aivenio/design-system", () => {
+jest.mock("@aivenio/aquarium", () => {
   return {
     __esModule: true,
-    ...jest.requireActual("@aivenio/design-system"),
+    ...jest.requireActual("@aivenio/aquarium"),
 
     Icon: () => {
       return <div data-testid={"ds-icon"}></div>;
     },
   };
 });
 
+const linkText = "Go to your profile page";
+
 describe("HeaderMenuLink.tsx", () => {
   // (icon is not needed for the test, Icon component mocked out)
   const mockIcon = "" as unknown as typeof data;
 
   describe("renders a default link with required props", () => {
-    beforeAll(() => {
+    beforeEach(() => {
       render(
         <HeaderMenuLink
           icon={mockIcon}
           href={"/myProfile"}
-          linkText={"Go to your profile page"}
+          linkText={linkText}
         />
       );
     });
 
-    afterAll(cleanup);
+    afterEach(cleanup);
 
     it(`renders a link with text dependent on a given property`, () => {
       const navLink = screen.getByRole("link", {
-        name: "Go to your profile page",
+        name: linkText,
       });
 
       expect(navLink).toBeVisible();
     });
 
-    it(`renders a href for that link dependent on a given  property`, () => {
+    it(`renders a href for that link dependent on a given property`, () => {
       const navLink = screen.getByRole("link", {
-        name: "Go to your profile page",
+        name: linkText,
       });
       expect(navLink).toHaveAttribute("href", "/myProfile");
     });
 
-    it(`renders a Tooltip with an Icon, both hidden for assistive technology`, () => {
+    it(`renders an Icon`, async () => {
+      const icon = screen.getByTestId("ds-icon");
+
+      expect(icon).toBeVisible();
+    });
+
+    it(`triggers the rendering of a Tooltip on mouse hover`, async () => {
       const navLink = screen.getByRole("link", {
-        name: "Go to your profile page",
+        name: linkText,
       });
-      const tooltip = within(navLink).getByTestId("tooltip");
-      const icon = within(tooltip).getByTestId("ds-icon");
+      const queryForToolTip = () => screen.queryByRole("tooltip");
+
+      expect(queryForToolTip()).toBeNull();
+
+      await userEvent.hover(navLink);
+
+      expect(queryForToolTip()).toBeVisible();
+      expect(queryForToolTip()).toHaveTextContent(linkText);
+
+      // Assert the tooltip disappears after hover event stops
+      cleanup();
```

Comment:
This test will always be 🟢  :D because `cleanup` removes the whole DOM. Doing a `userEvent. unhover` should do the trick and imitate the user behaviour best!

Reply:
Yes, this was a little bit of brute force haha! `unhover` works, as does `tab` for the keyboard test.

Reply:
[f928eda](https://github.com/aiven/klaw/pull/329/commits/f928eda4e39a0be17ae77134252bb09df0ceac0e)